ecommerce/api.py

```diff
@@ -118,13 +119,17 @@ def generate_checkout_payload(request):
         total_price += line_item.discounted_price
 
     if total_price == 0:
+        fulfill_completed_order(
+            order, {"amount": 0, "data": {"reason": "No payment required"}}, basket
+        )
         return {
             "no_checkout": True,
-            "response": fulfill_completed_order(
-                order,
-                {"amount": 0, "data": {"reason": "No payment required"}},
-                basket,
-                USER_MSG_TYPE_PAYMENT_ACCEPTED_NOVALUE,
```

Comment:
`fulfill_completed_order ` was being called with wrong number of args, so i had to refactor it to make it work the way it should.